projects/lib/sort-modules.js

```diff
@@ -0,0 +1,283 @@
+function cleanForSortingNumbers( values ){
+    if( Array.isArray( values ) && values.length > 0 ) {
+        const cleanedList = [];
+        const listToClean = [...values];
+
+        for (let i = 0; i < listToClean.length; i++) {
+            if( typeof listToClean[i] === 'number' ){
+                cleanedList.push(listToClean[i]);
+            }
+        }
+
+        return cleanedList;
+    }
+
+    return values;
+}
+
+
+function bubbleSortNumbers( values, order = 'ASC' ){
+
+    if( Array.isArray( values ) && values.length > 1 ) { // We check that valuesToOrder is an array and has more than one element
+        const valuesToOrder = cleanForSortingNumbers(values);
+
+        let isSwapped = true;
+
+        do {
+            isSwapped = false;
+
+            if ( order === 'ASC' ) {
+                for (let i = 0; i < valuesToOrder.length - 1; i++) {
+                    if (valuesToOrder[i] > valuesToOrder[i + 1]) {
+                        [ valuesToOrder[i], valuesToOrder[i + 1] ] = [ valuesToOrder[i + 1], valuesToOrder[i] ];
+                        isSwapped = true;
+                    }
+                }
+            } else if ( order === 'DESC' ) {
+                for (let i = 0; i < valuesToOrder.length - 1; i++) {
+                    if (valuesToOrder[i] < valuesToOrder[i + 1]) {
+                        [ valuesToOrder[i], valuesToOrder[i + 1] ] = [ valuesToOrder[i + 1], valuesToOrder[i] ];

```

Comment:
Try to optimize, so that based on the variable `order` you can execute the `for-loop` with _major_ or _minor_ (rows 31 and 38) **only once**. 
There is currently some **redundant code**.

Reply:
Optimized, thanks.